apps/web-giddh/src/app/shared/header/components/custom-fields/custom-fields.component.ts

```diff
@@ -0,0 +1,324 @@
+import { BreakpointObserver } from "@angular/cdk/layout";
+import { Component, OnDestroy, OnInit, TemplateRef } from "@angular/core";
+import { FormArray, FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { cloneDeep } from "apps/web-giddh/src/app/lodash-optimized";
+import { GroupService } from "apps/web-giddh/src/app/services/group.service";
+import { ToasterService } from "apps/web-giddh/src/app/services/toaster.service";
+import { IOption } from "apps/web-giddh/src/app/theme/ng-virtual-select/sh-options.interface";
+import { BsModalRef, BsModalService } from "ngx-bootstrap/modal";
+import { ReplaySubject } from "rxjs";
+import { takeUntil } from "rxjs/operators";
+
+@Component({
+    selector: 'custom-fields',
+    templateUrl: './custom-fields.component.html',
+    styleUrls: [`./custom-fields.component.scss`]
```

Comment:
Consider adding `ChangeDetectionStrategy.OnPush` for new components (consider doing unit testing after the change)